mdal/mdal_data_model.cpp

```diff
@@ -332,3 +337,293 @@ size_t MDAL::Mesh::faceVerticesMaximumCount() const
+void MDAL::DateTime::setWithGregorianJulianCalendarValues( MDAL::DateTime::DateTimeValues values )
+{
+  //https://quasar.as.utexas.edu/BillInfo/JulianDatesG.html
+
+  int C = 0;
+  int Y = values.year;
+  int M = values.month;
+  int D = values.day;
+  if ( M <= 2 )
+  {
+    Y--;
+    M += 12;
+  }
+  if ( values.year > 1582 ||
+       ( values.year == 1582 && ( values.month > 10 || ( values.month == 10 && values.day >= 15 ) ) ) ) //gregorian calendar
+  {
+    int A = Y / 100;
+    int B = A / 4;
+    C = 2 - A + B;
+  }
+
+  int E = int( 365.25 * ( Y + 4716 ) );
+  int F = int( 30.6001 * ( M + 1 ) );
+  double julianDay = C + D + E + F - 1524.5;
+
+  mJulianTime = int64_t( julianDay * 24 * 3600 * 1000 +
+                         ( values.hours ) * 3600 * 1000 + //the formula set the day with hours at 00h
+                         values.minutes * 60 * 1000 +
+                         values.seconds * 1000 );
+}
+
+std::string MDAL::DateTime::toString( MDAL::DateTime::DateTimeValues values ) const
+{
+  std::string yearStr = std::to_string( values.year );
+
+  int miliseconds = int( ( values.seconds - int( values.seconds ) ) * 1000 + 0.5 );
+  std::string msStr;
+  if ( miliseconds > 0 )
+  {
+    if ( miliseconds < 10 )
+      msStr = prependZero( std::to_string( miliseconds ), 3 );
+    else if ( miliseconds < 100 )
+      msStr = prependZero( std::to_string( miliseconds ), 2 );
+
+    msStr = std::string( "," ).append( msStr );
+  }
+
+  std::string strDateTime = prependZero( std::to_string( values.year ), 4 ) + "-" +
+                            prependZero( std::to_string( values.month ), 2 ) + "-" +
+                            prependZero( std::to_string( values.day ), 2 ) + "T" +
+                            prependZero( std::to_string( values.hours ), 2 ) + ":" +
+                            prependZero( std::to_string( values.minutes ), 2 ) + ":" +
+                            prependZero( std::to_string( int( values.seconds ) ), 2 ) +
+                            msStr; /// TODO use another way to translate seconds with fraction aprt
+
+  return strDateTime;
+}
+
+MDAL::Duration MDAL::DateTime::operator-( const MDAL::DateTime &other ) const
+{
+  if ( !mValid || !other.mValid )
+    return Duration();
+  return Duration( mJulianTime - other.mJulianTime );
+}
+
+MDAL::Duration::Duration(): mDuration( 0 )
+{}
+
+MDAL::Duration::Duration( double duration, MDAL::Duration::Unit unit )
+{
+  switch ( unit )
+  {
+    case MDAL::Duration::milliseconds:
+      mDuration = int64_t( duration );
+      break;
+    case MDAL::Duration::seconds:
+      mDuration = int64_t( duration * 1000 );
+      break;
+    case MDAL::Duration::minutes:
+      mDuration = int64_t( duration * 60 * 1000 );
+      break;
+    case MDAL::Duration::hours:
+      mDuration = int64_t( duration * 60 * 60 * 1000 );
+      break;
+    case MDAL::Duration::days:
+      mDuration = int64_t( duration * 24 * 60 * 60 * 1000 );
+      break;
+    case MDAL::Duration::weeks:
```

Comment:
do we need weeks?

Reply:
Here, there are the time units which are fix in any context (as opposed to month and year). I don't know if we need now, but it can be used if needed...